syntax = "proto3";


service ClientBaseStation {
    
    rpc checkConnection (CheckConnection) returns (ConnectionStats){}
    rpc checkSTMConnection (CheckConnection) returns (ConnectionStats){}
    rpc startSTMSampling (OrderSTM) returns (ConnectionStats) {}
    rpc stopSTMSampling (OrderSTM) returns (ConnectionStats) {}
    rpc sendSTMData (Void) returns (stream STMData) {}
  }
  
  message OrderSTM{
    string order= 1;

  }
  message STMData {
    string data=1;
  }
  message CheckConnection{
    string stats = 1;
  }
  
  message ConnectionStats{
    string stats = 1;
  }

  message Void{

  }